Analysis/HistHelper.py

```diff
@@ -0,0 +1,191 @@
+        self.df = self.df.Define("res1b", f"nSelBtag == 1")
+        self.df = self.df.Define("res2b", f"nSelBtag == 2")
+
+    def defineQCDRegions(self):
+        tau2_iso_var = f"tau2_idDeepTau{self.deepTauYear()}{self.deepTauVersion}VSjet"
+        self.df = self.df.Define("OS", "tau1_charge*tau2_charge < 0")
+        self.df = self.df.Define("Iso", f"{tau2_iso_var} >= {Utilities.WorkingPointsTauVSjet.Medium.value}")
+        self.df = self.df.Define("AntiIso", f"{tau2_iso_var} >= {Utilities.WorkingPointsTauVSjet.VVVLoose.value} && !Iso")
+        self.df = self.df.Define("OS_Iso", f"OS && Iso")
+        self.df = self.df.Define("SS_Iso", f"!OS && Iso")
+        self.df = self.df.Define("OS_AntiIso", f"OS && AntiIso")
+        self.df = self.df.Define("SS_AntiIso", f"!OS && AntiIso")
+
+    def deepTauYear(self):
+        return deepTauYears[self.deepTauVersion]
+
+    def selectTrigger(self, trigger):
+        self.df = self.df.Filter(trigger)
+
+    def CreateColumnTypes(self):
+        colNames = [str(c) for c in self.df.GetColumnNames()]
+        entryIndexIdx = colNames.index("entryIndex")
+        colNames[entryIndexIdx], colNames[0] = colNames[0], colNames[entryIndexIdx]
+        self.colNames = colNames
+        self.colTypes = [str(self.df.GetColumnType(c)) for c in self.colNames]
+
+    def __init__(self, df, deepTauVersion='v2p1'):
+        self.df = df
+        self.colNames=[]
+        self.colTypes=[]
+        self.deepTauVersion = deepTauVersion
+        self.bTagWP = 2
+        self.var_list = []
+        self.CreateColumnTypes()
+
+
+    def GetEventsFromShifted(self, df_central):
+        df_final = df_central.Filter(""" std::find ( analysis::GetEntriesVec().begin(), analysis::GetEntriesVec().end(),
+                                     entryIndex ) != analysis::GetEntriesVec().end()""")
+        self.df=df_final
+
+    def CreateFromDelta(self,var_list,central_columns,central_col_types):
+        for var_idx,var_name in enumerate(self.colNames):
+            if not var_name.endswith("Diff"):
+                continue
+            var_name_forDelta = var_name.removesuffix("Diff")
+            central_col_idx = central_columns.index(var_name_forDelta)
+            if central_columns[central_col_idx]!=var_name_forDelta:
+                print("ERRORE!")
+            self.df = self.df.Define(f"{var_name_forDelta}", f"""analysis::FromDelta({var_name},
+                                     analysis::GetEntriesMap()[entryIndex]->GetValue<{col_type_dict[self.colTypes[var_idx]]}>({central_col_idx}) )""")
+            var_list.append(f"{var_name_forDelta}")
+        for central_col_idx,central_col in enumerate(central_columns):
+            if central_col in var_list or central_col in self.colNames: continue
+            if central_col != 'channelId' : continue # this is for a bugfix that I still haven't figured out !!
```

Comment:
I put this because it has problems for the column `sample_name` 
and I think it will have problems also for vectorial columns. So I need to figure out:
- if we need some columns in particular from central, except for channelID - but at least for now we don't
- even if we don't, for completeness I would also add the possibility to take central columns whatever type they are